src/backend/catalog/aclchk.c

```diff
@@ -224,16 +224,17 @@ restrict_and_check_grant(bool is_grant, AclMode avail_goptions, bool all_privs,
 	 * If we found no grant options, consider whether to issue a hard error.
 	 * Per spec, having any privilege at all on the object will get you by
 	 * here.
+	 * Superusers or on QE bypass all permission checking.
```

Comment:
For ranger, we should still do checks but for normal ACL checks, we should bypass all permission checking for  superusers. I should change the comment, and thanks.